<?xml version="1.0"?>
<launch>
    <!-- 红色图案检测器 -->
    <node pkg="landing_detection" 
          type="red_pattern_detector.py" 
          name="red_detector" 
          output="screen">
        
        <!-- 红色检测参数 (HSV) -->
        <param name="red_lower1_h" value="0" />
        <param name="red_lower1_s" value="100" />
        <param name="red_lower1_v" value="100" />
        <param name="red_upper1_h" value="10" />
        <param name="red_upper1_s" value="255" />
        <param name="red_upper1_v" value="255" />
        
        <param name="red_lower2_h" value="170" />
        <param name="red_lower2_s" value="100" />
        <param name="red_lower2_v" value="100" />
        <param name="red_upper2_h" value="180" />
        <param name="red_upper2_s" value="255" />
        <param name="red_upper2_v" value="255" />
        
        <!-- 白色检测参数 (HSV) -->
        <param name="white_lower_h" value="0" />
        <param name="white_lower_s" value="0" />
        <param name="white_lower_v" value="200" />
        <param name="white_upper_h" value="180" />
        <param name="white_upper_s" value="30" />
        <param name="white_upper_v" value="255" />
        
        <!-- 几何参数 -->
        <param name="min_circle_radius" value="30" />
        <param name="max_circle_radius" value="200" />
        <param name="min_contour_area" value="1000" />
        <param name="circularity_threshold" value="0.65" />
        <param name="red_area_threshold" value="0.6" />
        
        <!-- 话题重映射 -->
        <remap from="/iris_0/camera/image_raw" to="/iris_0/camera/image_raw" />
    </node>
    
</launch>
